platform-sdk/swirlds-jasperdb/src/test/java/com/swirlds/merkledb/files/DataFileCollectionTest.java

```diff
@@ -713,6 +702,108 @@ void mergeWorksAfterOpen(final FilesTestType testType) throws Exception {
+                        break;
+                }
+                // store in file
+                storedOffsets.put(i, fileCollection.storeDataItem(dataValue));
+            }
+            fileCollection.endWriting(0, count + 100).setFileCompleted();
+            count += 100;
+        }
+    }
+
+    /**
+     * This test emulates scenario in which compaction is interrupted by thread interruption. This event shouldn't be
+     * reported as an error in the logs.
+     */
+    @Test
+    public void testClosedByInterruptException() throws IOException {
+        // mock appender to capture the log statements
+        final MockAppender mockAppender = new MockAppender("testClosedByInterruptException");
+        Logger logger = (Logger) LogManager.getLogger(DataFileCollection.class);
+        mockAppender.start();
+        logger.addAppender(mockAppender);
+        final Path dbDir = tempFileDir.resolve("testClosedByInterruptException");
+        final String storeName = "testClosedByInterruptException";
+
+        // init file collection with some content to compact
+        final DataFileCollection<long[]> fileCollection =
+                new DataFileCollection<>(dbDir, storeName, FilesTestType.fixed.dataItemSerializer, null);
+        final LongListHeap storedOffsets = new LongListHeap(5000);
+        populateDataFileCollection(FilesTestType.fixed, fileCollection, storedOffsets);
+
+        // a flag to make sure that `compactFiles` th
+        AtomicBoolean closedByInterruptFromCompaction = new AtomicBoolean(false);
+
+        final Thread thread = new Thread(() -> {
+            List<DataFileReader<long[]>> allCompletedFiles = fileCollection.getAllCompletedFiles();
+            DataFileReader<long[]> spy = Mockito.spy(allCompletedFiles.get(0));
+            try {
+                AtomicInteger count = new AtomicInteger(0);
+                when(spy.getMetadata()).thenAnswer(invocation -> {
+                    // on the second call to getMetadata, we interrupt the thread
+                    if (count.getAndIncrement() == 1) {
+                        Thread.currentThread().interrupt();
+                    }
+
+                    return invocation.callRealMethod();
+                });
+
+                List<DataFileReader<long[]>> allCompletedFilesUpdated = new ArrayList<>(allCompletedFiles);
+                allCompletedFilesUpdated.set(0, spy);
+                fileCollection.compactFiles(storedOffsets, allCompletedFilesUpdated);
+            } catch (InterruptedException e) {
+                // we expect interrupted exception here
+                closedByInterruptFromCompaction.set(true);
+            } catch (IOException e) {
+                fail("Exception should not be thrown");
+            }
+            reset(spy);
+        });
+        thread.start();
+        try {
+            assertEventuallyTrue(
+                    () -> {
+                        try {
+                            Thread.sleep(100);
```

Comment:
This Thread.sleep() is probably redundant because of assertEventuallyTrue()

Reply:
Agreed. I removed it